api/pipeline/get.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2022 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package pipeline
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/go-vela/server/router/middleware/org"
+	"github.com/go-vela/server/router/middleware/pipeline"
+	"github.com/go-vela/server/router/middleware/repo"
+	"github.com/go-vela/server/router/middleware/user"
+	"github.com/sirupsen/logrus"
+)
+
+// swagger:operation GET /api/v1/pipelines/{org}/{repo}/{pipeline} pipelines GetPipeline
+//
+// Get a pipeline from the configured backend
+//
+// ---
+// produces:
+// - application/json
+// parameters:
+// - in: path
+//   name: org
+//   description: Name of the org
+//   required: true
+//   type: string
+// - in: path
+//   name: repo
+//   description: Name of the repo
+//   required: true
+//   type: string
+// - in: path
+//   name: pipeline
+//   description: Commit SHA for pipeline to retrieve
+//   required: true
+//   type: string
+// - in: query
+//   name: output
+//   description: Output string for specifying output format
+//   type: string
+// security:
+//   - ApiKeyAuth: []
+// responses:
+//   '200':
+//     description: Successfully retrieved the pipeline
+//     type: json
+//     schema:
+//       "$ref": "#/definitions/Pipeline"
+
+// GetPipeline represents the API handler to capture
+// a pipeline for a repo from the configured backend.
+func GetPipeline(c *gin.Context) {
+	// capture middleware values
+	o := org.Retrieve(c)
+	p := pipeline.Retrieve(c)
+	r := repo.Retrieve(c)
+	u := user.Retrieve(c)
+
+	// update engine logger with API metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Entry.WithFields
+	logrus.WithFields(logrus.Fields{
+		"org":      o,
+		"pipeline": p.GetCommit(),
+		"repo":     r.GetName(),
+		"user":     u.GetName(),
+	}).Infof("reading pipeline %s/%s", r.GetFullName(), p.GetCommit())
```

Comment:
with the given fields, the contents of `%s` don't provide any new information. should we drop it?

Reply:
responded in earlier comment 👍